### DISCUSSION TITLE
MPC55xx and 56xx family decompiler issue

### DISCUSSION THREAD

**Discussion by mcar5000** [NationalSecurityAgency/ghidra#3839] [Q&A] [answered]
Hello everyone, 
I want to share a doubt that is struggling with myself. I correctly handled the decompilation of a binary file from an MPC55xx Microcontroller retrieving all the functions and variables useful to analyze the entire code. Now for making a new analysis, I switched to the MPC56xx family, which should use the same core (the same instruction set) but a more big flash size. I was surprised that Ghidra is not starting the decompilation at all of this binary while it should be the same (in terms of family) as MPC55xx. Do you have any idea?

Thank you
M

**Comment by mcar5000** [comment, marked as answer]
#Update, the only part that does not seem to work, is the auto analysis. Decompiling manually, it works like a charm! Best tool ever!

**Reply by astrelsky** [reply]
Hmm, I'm not sure where that processor is so I can't provide a direct link. It seems like there may be an issue with the pattern constraints or they aren't being applied to the `MPC56xx` family. Here are the pattern files for MIPS as a reference to what I'm talking about.
https://github.com/NationalSecurityAgency/ghidra/tree/master/Ghidra/Processors/MIPS/data/patterns

**Comment by brianbri6** [comment]
> Hello everyone, I want to share a doubt that is struggling with myself. I correctly handled the decompilation of a binary file from an MPC55xx Microcontroller retrieving all the functions and variables useful to analyze the entire code. Now for making a new analysis, I switched to the MPC56xx family, which should use the same core (the same instruction set) but a more big flash size. I was surprised that Ghidra is not starting the decompilation at all of this binary while it should be the same (in terms of family) as MPC55xx. Do you have any idea?
> 
> Thank you M

any pointers on setting up MPC55xx bin file in Ghidra for a noob?

**Comment by oemotorsport** [comment]
Hello, can i contact with you by email or social network?

**Comment by mcar5000** [comment]
Hi, yes of course.
@mcar500 on telegram or mcar500 instagram
M

Il giorno mer 24 gen 2024 alle 20:27 oemotorsport ***@***.***>
ha scritto:
